app/src/main/java/com/weatherxm/ui/widgets/selectstation/SelectStationAdapter.kt

```diff
@@ -147,6 +165,49 @@ class SelectStationAdapter(private val stationListener: (UIDevice) -> Unit) :
                 ), Weather.getPrecipitationPreferredUnit()
             )
         }
+
+        private fun setCardStroke(item: UIDevice) {
+            /**
+             * If the UIDevice has an error alert or an error metric then the stroke should be error
+             * or if there are warning alerts or warning metrics then the stroke should be warning
+             * otherwise clear the stroke
+             */
+            if (item.hasErrors() || item.hasErrorMetrics()) {
+                binding.root.setCardStroke(R.color.error, 2)
+            } else if (item.alerts.isNotEmpty() || item.hasWarningMetrics()) {
+                binding.root.setCardStroke(R.color.warning, 2)
+            } else {
+                binding.root.strokeWidth = 0
+            }
+        }
+
+        private fun setAlerts(item: UIDevice) {
+            if (item.alerts.size > 1) {
+                if (item.hasErrors()) {
+                    binding.issueChip.errorChip()
+                } else {
+                    binding.issueChip.warningChip()
+                }
+                binding.issueChip.text =
+                    itemView.context.getString(R.string.issues, item.alerts.size)
+            } else {
+                when (item.alerts[0]) {
+                    DeviceAlert.createWarning(DeviceAlertType.LOW_BATTERY) -> {
+                        binding.issueChip.lowBatteryChip()
+                    }
+                    DeviceAlert.createError(DeviceAlertType.OFFLINE) -> {
+                        binding.issueChip.offlineChip()
```

Comment:
Can we encapsulate somewhere this functionality to avoid duplication with the devices' adapter?
In general, in the iOS app, we declare an extension of the UIDevice model (`DeviceDetails` in iOS codebase) which generates this kind of stuff